evap/results/tests/test_views.py

```diff
@@ -772,120 +771,212 @@ def test_textanswer_visibility_for_manager(self):
         self.assertNotIn(".responsible_contributor_additional_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_additional_orig_hidden.", page)
+        self.assertNotIn(".responsible_contributor_additional_orig_deleted.", page)
+        self.assertNotIn(".general_additional_orig_published.", page)
+        self.assertNotIn(".general_additional_orig_deleted.", page)
 
-    def test_textanswer_visibility_for_responsible_contributor(self):
-        page = self.app.get("/results/semester/1/evaluation/1", user="responsible_contributor@institution.example.com")
-        self.assertIn(".general_orig_published.", page)
-        self.assertNotIn(".general_orig_hidden.", page)
-        self.assertNotIn(".general_orig_published_changed.", page)
-        self.assertIn(".general_additional_orig_published.", page)
-        self.assertNotIn(".general_additional_orig_hidden.", page)
-        self.assertIn(".general_changed_published.", page)
-        self.assertNotIn(".contributor_orig_published.", page)
-        self.assertNotIn(".contributor_orig_private.", page)
-        self.assertIn(".responsible_contributor_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_orig_hidden.", page)
-        self.assertNotIn(".responsible_contributor_orig_published_changed.", page)
-        self.assertIn(".responsible_contributor_changed_published.", page)
-        self.assertIn(".responsible_contributor_orig_private.", page)
-        self.assertNotIn(".responsible_contributor_orig_notreviewed.", page)
-        self.assertIn(".responsible_contributor_additional_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_additional_orig_hidden.", page)
+    def test_textanswer_visibility_for_student(self):
+        # NOTE: default values for students are: view_general_results=ratings, view_contributor_results=ratings
+        general_views = ["full", "ratings"]
+        contributor_views = ["full", "ratings", "personal"]
+
+        for general_view in general_views:
+            page = self.app.get(
+                f"/results/semester/1/evaluation/1?view_general_results={general_view}&view_contributor_results=ratings",
+                user="student@institution.example.com",
+            )
+            self.helper_static(page)
+
+        for contributor_view in contributor_views:
+            page = self.app.get(
+                f"/results/semester/1/evaluation/1?view_general_results=ratings&view_contributor_results={contributor_view}",
+                user="student@institution.example.com",
+            )
+            self.helper_static(page)
+
+
+class TestResultsTextanswerVisibilityForResponsible(WebTest):
+    fixtures = ["minimal_test_data_results"]
+
+    @classmethod
+    def setUpTestData(cls):
+        cache_results(Evaluation.objects.get(id=1))
+
+    def helper_static(self, page):
+
+        self.assertNotIn(".general_orig_deleted.", page)
```

Comment:
Instead of defining one function per test case, I'd propose you simply store a list of all text answers. In your test, for one page, you then pass a list of expected visible text answers to a shared helper function. The helper asserts that the expected answers are in the page, and that all answers that are not expected are not in the page. That way, you should reduce the duplication of the `helper_static` method, the `for` loops, and the repeated logic of calling the "static" and the "dynamic" helper

Reply:
(In the end, a test could probably look like:

```helper
my_test_helper("full", "full", [".some_textanswer", ".some_other_textanswer"])
my_test_helper("full", "personal", [".some_textanswer", ".some_other_textanswer"])
...
```
)